multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/HostValidator.java

```diff
@@ -23,11 +29,31 @@ public boolean isValid(Object routePart, final Map<String, Object> context) {
             return false;
         }
         if (shouldApplyNamespace(context) && getNamespace() != null) {
+            if (shouldApplyNamespaceAsSuffix(context)) {
+                return isHostWithNamespaceAndSuffixValid((String) routePart);
+            }
             return ((String) routePart).startsWith(getNamespace());
         }
         return true;
     }
 
+    private boolean isHostWithNamespaceAndSuffixValid(String routePart) {
+        if (routePart.endsWith(getNamespace())) {
+            return true;
+        } else if (routePart.endsWith(SystemParameters.IDLE_HOST_SUFFIX)) {
+            String routeStringWithoutIdleSuffix = routePart.substring(0, routePart.lastIndexOf(SystemParameters.IDLE_HOST_SUFFIX));
```

Comment:
routeStringWithoutIdleSuffix if a bad name for all of the else statements; also a lot of the same 2 lines - maybe extract in a method 'boolean containsNamespaceBeforeSuffix(routePart, suffix)